src/InfoViz/Native/ParallelCoordinates/index.js

```diff
@@ -841,6 +850,24 @@ function parallelCoordinate(publicAPI, model) {
     }
   });
 
+  if (model.provider.isA('DataUpdateProvider')) {
+    model.updateSubscription = model.provider.subscribeToDataUpdate(
+      'histogram2d',
+      data => {
+        Object.keys(data).forEach(xName => {
+          Object.keys(data[xName]).forEach(yName => {
+            const histoPayload = data[xName][yName][0];
+            if (histoPayload.role && histoPayload.role.type && histoPayload.role.type === 'alldata') {
+              model.provider.setHistogram2D(xName, yName, histoPayload);
```

Comment:
It seems very odd that PC is connecting two data providers together like this. It looks like an artifact of the Histogram2D provider being designed before the DataUpdate provider. However, I don't know how to resolve it.